src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt

```diff
@@ -33,9 +36,19 @@ class AuthenticationFilter(private val sessionService: SessionService) : Contain
             unauthorizedUser(requestContext)
 
         try {
-            Jwts.parserBuilder()
-                .setSigningKey(sessionService.secretKey)
+            val claims: Claims = Jwts.parserBuilder()
+                .setSigningKey(authService.secretKey)
                 .build()
+                .parseClaimsJws(token).body
+
+            val employeeId = (claims["employeeId"] as? Int)?.toLong()
+            val companyId = (claims["companyId"] as? Int)?.toLong()
+            val role = enumValueOf<Role>(claims["role"] as String)
```

Comment:
in order to make sure that you don't have a case issue ("admin" vs "ADMIN") it's always better to make sure that you use upper/lower case.
In case you get something like "Admin", and then it won't be converted to the enum, because in the enum it is lower case (for example), the case has to match